### PR TITLE
[BE][optim] abstract out docstrings, add differentiable docs

### PR DESCRIPTION
**Pull request by janeyx99**
1. abstract out common doc strings --> I'm sure there are more, but let this be a first step.
2. Add differentiable docs to those who are actually differentiable

### CODE REVIEW

File: torch/optim/optimizer.py [hunk 1]
```diff
@@ -67,6 +67,27 @@ def _default_to_foreach(tensorlists: List[List[torch.Tensor]], differentiable: b
     )
 
 
+# Common doc strings among optimizers
```
Comment (albanD) [inline torch/optim/optimizer.py:70]: nit: these should most likely be private ;)

**Review by albanD — CHANGES_REQUESTED** [review]

File: torch/optim/optimizer.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/optimizer.py:70]: done

Comment (janeyx99) [reply torch/optim/optimizer.py:70]: i did not bother reordering imports

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**